src/main/java/com/hierynomus/msfscc/fsctl/FsCtlPipeWaitRequest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C)2016 - SMBJ Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hierynomus.msfscc.fsctl;
+
+import com.hierynomus.protocol.commons.buffer.Buffer;
+
+import java.nio.charset.StandardCharsets;
+
+/**
+ * [MS-FSCC] 2.3.29 FSCTL_PIPE_WAIT Request
+ */
+public class FsCtlPipeWaitRequest {
+    private String name;
+    private long timeout;
+    private boolean timeoutSpecified;
+
+    public FsCtlPipeWaitRequest(String name, long timeout, boolean timeoutSpecified) {
+        this.name = name;
+        this.timeout = timeout;
+        this.timeoutSpecified = timeoutSpecified;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public long getTimeout() {
+        return timeout;
+    }
+
+    public void write(Buffer buffer) {
+        buffer.putUInt64(timeout / 100L);
+
+        int nameLengthPos = buffer.wpos();
+        buffer.putUInt32(0);
```

Comment:
Is there a method that I can use to calculate the length? I know it's essentially `string.length() * 2` for UTF-16, but I don't think it's a good idea to duplicate that calculation everywhere.
I wrote the code this way to keep things decoupled. Buffer#putString encodes the string, FsCtlPipeWaitRequest#write doesn't need to care how it's done exactly. It's length value will always be correct.